From a67390c72369d7c17ca548343153f8bfed1bf1e1 Mon Sep 17 00:00:00 2001
Message-Id: <a67390c72369d7c17ca548343153f8bfed1bf1e1.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: =?UTF-8?q?Ville=20Syrj=C3=A4l=C3=A4?= <ville.syrjala@linux.intel.com>
Date: Sat, 28 Jun 2014 02:04:07 +0300
Subject: [PATCH 188/312] FROM_UPSTREAM [VPG]: drm/i915: Add chv_power_wells[]
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add chv_power_wells[] so we can start to build up the power well support
for chv. Just the "always on" well there initialy.

Change-Id: I710ce6012e611d6bb020dd19d8a8be752138280d
Signed-off-by: Ville Syrjälä <ville.syrjala@linux.intel.com>
Upstream-repo: git://anongit.freedesktop.org/drm-intel
Upstream-commit: a1e633f780e86e7071d7cb547ed82666145fd4a4
Upstream-as-of-tag: <not available>
---
 drivers/gpu/drm/i915/intel_pm.c |   12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 797c1ba..0d19c27 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -7144,6 +7144,16 @@ static struct i915_power_well vlv_power_wells[] = {
 	},
 };
 
+static struct i915_power_well chv_power_wells[] = {
+	{
+		.name = "always-on",
+		.always_on = 1,
+		.domains = VLV_ALWAYS_ON_POWER_DOMAINS,
+		.ops = &i9xx_always_on_power_well_ops,
+	},
+};
+
+
 #define set_power_wells(power_domains, __power_wells) ({		\
 	(power_domains)->power_wells = (__power_wells);			\
 	(power_domains)->power_well_count = ARRAY_SIZE(__power_wells);	\
@@ -7165,6 +7175,8 @@ int intel_power_domains_init(struct drm_i915_private *dev_priv)
 	} else if (IS_BROADWELL(dev_priv->dev)) {
 		set_power_wells(power_domains, bdw_power_wells);
 		hsw_pwr = power_domains;
+	} else if (IS_CHERRYVIEW(dev_priv->dev)) {
+		set_power_wells(power_domains, chv_power_wells);
 	} else if (IS_VALLEYVIEW(dev_priv->dev)) {
 		set_power_wells(power_domains, vlv_power_wells);
 	} else {
-- 
1.7.9.5

